### PR TITLE
Compress QCOW2 when shrinking

### PR DESCRIPTION
**Pull request by CMeza99**
Seems like one would want a compressed QCOW2 image if the volume is being shrunk.

### CODE REVIEW

File: bootstrapvz/plugins/minimize_size/tasks/shrink.py [hunk 1]
```diff
@@ -70,6 +70,8 @@ class ShrinkVolumeWithQemuImg(Task):
     @classmethod
     def run(cls, info):
         tmp_name = os.path.join(info.workspace, 'shrunk.' + info.volume.extension)
-        log_check_call(
-            ['qemu-img', 'convert', '-O', info.volume.extension, info.volume.image_path, tmp_name])
+        shrink_cmd = ['qemu-img', 'convert', '-O', info.volume.extension, info.volume.image_path, tmp_name]
+        if info.volume.extension == 'qcow2':
+            shrink_cmd.insert(4, '-c')
```
Comment (andsens) [inline bootstrapvz/plugins/minimize_size/tasks/shrink.py]: I think you need to add some explanation for what this does `insert(4, '-c')` might be a bit... terse :-)

Too bad qemu-img doesn't have long options.

**Comment by CMeza99** [comment]
Yup, that definitely should be clarified. I added some comments, will that suffice?

**Comment by andsens** [comment]
Plenty sufficient :-)  
Merged.

**Comment by mengelmann** [comment]
I know it's merged already, but shouldn't the compression be configurable? There is a performance impact each time when reading from a compressed qcow2 image.

**Comment by andsens** [comment]
I was unaware. @CMeza99?

**Pull request merged**